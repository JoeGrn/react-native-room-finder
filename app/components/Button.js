import React from 'react'
import { ViewPropTypes, TouchableOpacity, StyleSheet, Text, View } from 'react-native'
import PropTypes from 'prop-types'

const getColorScheme = colorScheme => {
    switch (colorScheme) {
        case 'white':
            return {
                colorSchemeOuter: styles.whiteButton,
                colorSchemeInner: styles.blackText,
            }
        default:
            return {
                colorSchemeOuter: styles.greenButton,
            }
    }
}

const Button = ({
    title,
    onPress,
    style,
    textStyle,
    disabled,
    accessibilityLabel,
    testID,
    block,
    colorScheme,
}) => {
    const { colorSchemeOuter, colorSchemeInner } = getColorScheme(colorScheme)
    return (
        <View style={[block && styles.block]}>
            <TouchableOpacity
                onPress={onPress}
                disabled={disabled}
                style={[styles.buttonStyle, colorSchemeOuter, style]}
                accessibilityLabel={accessibilityLabel}
                testID={testID}
            >
                <Text style={[styles.textStyle, textStyle, colorSchemeInner]}>{title}</Text>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    buttonStyle: {
        borderRadius: 5,
        borderWidth: 2,
    },
    textStyle: {
        alignSelf: 'center',
        color: '#fff',
        fontSize: 16,
        fontWeight: 'bold',
        padding: 10,
    },
    whiteButton: {
        backgroundColor: '#fff',
        borderColor: 'black',
    },
    greenButton: {
        borderColor: '#68A51D',
        backgroundColor: '#68A51D',
    },
    blackText: {
        color: '#000',
    },
    block: {
        flex: 1,
    },
})

Button.defaultProps = {
    disabled: false,
}

Button.propTypes = {
    title: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    style: ViewPropTypes.style,
    textStyle: PropTypes.object,
    colorScheme: PropTypes.string,
    disabled: PropTypes.bool,
    block: PropTypes.bool,
}

export default Button
