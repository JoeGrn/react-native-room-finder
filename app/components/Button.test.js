import Button from './Button'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import React from "react";

describe('<Button/>',() => {

    const onPress = jest.fn()

    beforeEach(() => {
        jest.clearAllMocks()
    })

    it('should render', () => {

        const { getByTestId, getByText, queryByTestId, asJSON } = render(
            <Button title="my button" onPress={onPress}/>
        )

        expect(asJSON()).toMatchSnapshot();

    })

})
