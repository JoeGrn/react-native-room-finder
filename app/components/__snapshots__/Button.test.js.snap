// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <
      disabled={false}
      style={
        Array [
          Object {
            "borderRadius": 5,
            "borderWidth": 2,
          },
          Object {
            "backgroundColor": "#68A51D",
            "borderColor": "#68A51D",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "bold",
              "padding": 10,
            },
            undefined,
            undefined,
          ]
        }
      >
        my button
      </Text>
    </>
  </View>
</View>
`;
